src/audio/mixer/mixer.c

```diff
@@ -179,6 +179,32 @@ static int mixer_reset(struct processing_module *mod)
 	return 0;
 }
 
+/* init and calculate the aligned setting for available frames and free frames retrieve*/
+static inline void mixer_set_frame_alignment(struct audio_stream __sparse_cache *source)
+{
+#if XCHAL_HAVE_HIFI3 || XCHAL_HAVE_HIFI4
+
+	/* Xtensa intrinsics ask for 8-byte aligned. 5.1 format SSE audio
+	 * requires 16-byte aligned.
+	 */
+	const uint32_t byte_align = source->channels == 6 ? 16 : 8;
```

Comment:
@singalsu @dbaluta @andrula-song we need to come up with some arch SIMD utility macros/inline for this so we can use on all processing.

Reply:
would #define XTENSA_BYTE_ALIGN(channels) (channels) == 6 ? 16 : 8 in  numbers.h okay?

Reply:
We should have an architecture generic macro in numbers.h e.g.
```c
#if ARCH_XTENSA
#if HIFI3
#define ARCH_SIMD_ALIGN(chan) 
#elif HIFI4
#define ARCH_SIMD_ALIGN(chan) 
#endif
#elif ARCH_other
#define ARCH_SIMD_ALIGN(chan)
#endif
```

Reply:
ok, I will create a new PR to add this macro and replace all such usage in our SOF.

Reply:
ok, lets merge the new PR first and then redo this PR to use it.

Reply:
and the xtensa test result is below, the functions with '_1' is the new one, can improve about 15% cycles than the original ones.
![Screenshot from 2023-02-02 14-43-29](https://user-images.githubusercontent.com/93516384/217131102-c9a87417-17c0-44a7-8195-bf0b73d82244.png)